10_temperature_object/temperature.rb

```diff
@@ -1,39 +1,40 @@
 # frozen_string_literal: true
 
 class Temperature
-  def initialize(hash = {})
-    @hash = hash
+  def initialize(hash)
+    @ftemp = hash[:f]
+    @ctemp = hash[:c]
+    @ftemp = (@ctemp.to_f * 9 / 5) + 32 unless @ctemp.nil?
+    @ctemp = (@ftemp - 32) * 5 / 9 unless @ftemp.nil?
+  end
+
+  def in_fahrenheit
+    @ftemp
+  end
+
+  def in_celsius
+    @ctemp
   end
 
   def self.from_celsius(temp)
-    Temperature.new({ c: temp })
+    self.new(:c => temp)
   end
 
   def self.from_fahrenheit(temp)
-    Temperature.new({ f: temp })
+    self.new(:f => temp)
   end
+end
 
-  def in_fahrenheit
-    if @hash.key?(:f)
-      @hash[:f]
-    elsif @hash.key?(:c)
-      @hash[:c] * 9.to_f / 5 + 32
-    elsif @temp.is_a? Numeric
-      @temp * 9.to_f / 5 + 32
-    else
-      @temp
-    end
+class Celsius < Temperature
+  def initialize(temp)
+    @ctemp = temp
+    @ftemp = (@ctemp.to_f * 9 / 5) + 32 unless @ctemp.nil?
   end
+end
 
-  def in_celsius
-    if @hash.key?(:c)
-      @hash[:c]
-    elsif @hash.key?(:f)
-      (@hash[:f] - 32) * 5.to_f / 9
-    elsif @temp.is_a? Numeric
-      (@temp - 32) * 5.to_f / 9
-    else
-      @temp
-    end
+class Fahrenheit < Temperature
```

Comment:
What are these used for?

Reply:
Subclasses used to build temperature in subclass

Reply:
```
# frozen_string_literal: true

class Temperature
  def self.from_celsius(temp)
    Celsius.new(temp)
  end

  def self.from_fahrenheit(temp)
    Fahrenheit.new(temp)
  end

  def self.new(args)
    if args.key(:f)
      from_fahrenheit(args[:f])
    else
      from_celsius(args[:c])
    end
  end
end

class Celsius
  def initialize(temp)
    @temp = temp
  end

  def in_celsius
    @temp
  end

  def in_fahrenheit
    (@temp * 9.0 / 5.0) + 32
  end
end

class Fahrenheit
  def initialize(temp)
    @temp = temp
  end

  def in_celsius
    (@temp - 32) * (5.0 / 9.0)
  end

  def in_fahrenheit
    @temp
  end
end
```